cpp/src/arrow/ipc/metadata-internal.cc

```diff
@@ -251,6 +251,9 @@ Status ConcreteTypeFromFlatbuffer(flatbuf::Type type, const void* type_data,
       return Status::OK();
     case flatbuf::Type_Decimal: {
       auto dec_type = static_cast<const flatbuf::Decimal*>(type_data);
+      if ((dec_type->precision() < 1) || (dec_type->precision() > 38)) {
```

Comment:
Could we make (1, 38) a constexpr in either type.h or basic_decimal.h. I see them in gandiva's decimal_type_util.h, bring them back in arrow/